eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/ImpTest.hs

```diff
@@ -761,22 +761,19 @@ fixupFees tx = do
   certState <- getsNES $ nesEsL . esLStateL . lsCertStateL
   kpSpending <- lookupKeyPair =<< freshKeyHash
   kpStaking <- lookupKeyPair =<< freshKeyHash
-  rootTxIn <- fst <$> lookupImpRootTxOut
   nativeScriptKeyPairs <- impNativeScriptKeyPairs tx
   let
     nativeScriptKeyWits = Map.keysSet nativeScriptKeyPairs
-    txWithRoot = tx & bodyTxL . inputsTxBodyL %~ Set.insert rootTxIn
-    consumedValue = consumed pp certState utxo (txWithRoot ^. bodyTxL)
-    producedValue = produced pp certState (txWithRoot ^. bodyTxL)
+    consumedValue = consumed pp certState utxo (tx ^. bodyTxL)
+    producedValue = produced pp certState (tx ^. bodyTxL)
     changeBeforeFee = consumedValue <-> producedValue
     changeBeforeFeeTxOut =
       mkBasicTxOut
         (mkAddr (kpSpending, kpStaking))
         changeBeforeFee
   let
-    txNoWits =
-      txWithRoot & bodyTxL . outputsTxBodyL %~ (:|> changeBeforeFeeTxOut)
-    outsBeforeFee = txWithRoot ^. bodyTxL . outputsTxBodyL
+    txNoWits = tx & bodyTxL . outputsTxBodyL %~ (:|> changeBeforeFeeTxOut)
+    outsBeforeFee = tx ^. bodyTxL . outputsTxBodyL
```

Comment:
I can see it's correct to remove this (because i tested it myself out of disbelief), but I can't explain why this is the case! The root is in the UTXO, so how come it's all still balancing out?

Reply:
Because there is now a separate step that does just that with `addRootTxIn` 
https://github.com/IntersectMBO/cardano-ledger/blob/bf878d234657c50c4bdc748d324460b92d05a522/eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/ImpTest.hs#L803

Reply:
Oh right, I missed this. Thanks!